### PR TITLE
C++ API parity: at::Tensor::output_nr

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26332 C++ API parity: at::Tensor::requires_grad_
* #26251 C++ API parity: at::Tensor::detach
* #26217 C++ API parity: at::Tensor::version
* **#26216 C++ API parity: at::Tensor::output_nr**

Differential Revision: [D17427576](https://our.internmc.facebook.com/intern/diff/D17427576)

### CODE REVIEW

**Review by yf225 — APPROVED** [review]
Thanks a lot @pbelevich !
